packages/terra-consumer-nav/src/components/user-profile/UserProfile.jsx

```diff
@@ -13,31 +14,40 @@ const cx = classNames.bind(styles);
 
 const propTypes = {
   /**
-   * User name to be displayed in profile.
+   * User name to be displayed in profile if user is signed in.
    */
-  name: PropTypes.string.isRequired,
+  userName: PropTypes.string,
   /**
    * Avatar to be displayed in profile.
    */
-  avatar: PropTypes.oneOfType([
-    PropTypes.string,
-    PropTypes.element,
-  ]),
+  avatar: PropTypes.element,
   /**
    * A unique id set to the profile popup button that will be referred in profile popup.
    */
-  id: PropTypes.string.isRequired,
+  id: PropTypes.string,
   /**
    * The path signout button would redirect to.
    */
   signoutUrl: PropTypes.string.isRequired,
   /**
+   * The path to the login page.
+   */
+  signinUrl: PropTypes.string,
+  /**
+   * Determiniate of whether profile should render as signin link or profile button.
+   */
+  isSignIn: PropTypes.bool,
+  /**
    * The content of the each profile items.
    */
-  profileLinks: PropTypes.arrayOf(PropTypes.shape(
-    navElementShape)),
-    /**
-    */
+  profileLinks: PropTypes.arrayOf(
+    PropTypes.shape(
+      navElementShape,
```

Comment:
do we even use any of the navElementShape props here? ideally we only check the prop types in the component we are consuming them in. 
In my branch, i'm seeing if i have a single component that handles rendering the link (to handle all the target, type, react router, custom props, etc) if its the only component that needs all these props, then I will remove navElementShape

Reply:
yeah these seem to be passed straight through to the child, will remove validation from this.